marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mastering/default/merging.sjs

```diff
@@ -38,87 +37,87 @@ function main(content, options) {
   const collectionQuery = cts.collectionQuery(matchSummaryCollection);
-  const uriRangeQuery = cts.rangeQuery(urisPathReference, '=', uriToTakeActionOn);
-  let relatedMatchSummaries = cts.search(
-    cts.andQuery([
-      uriRangeQuery,
-      collectionQuery
-    ]),
-    ['unfiltered',cts.indexOrder(datahubCreatedOnRef, 'descending')]
-  );
-  let mostRecentMatchSummary = fn.head(relatedMatchSummaries);
-  let matchSummaryJson = mostRecentMatchSummary.toObject();
-  let uriActionDetails = matchSummaryJson.matchSummary.actionDetails[uriToTakeActionOn];
-  // If the action is merge, ensure we create the merge with the most URIs
-  if (uriActionDetails && uriActionDetails.action === 'merge') {
-    const urisQuery = cts.pathRangeQuery('//actionDetails/*/uris', '=', uriActionDetails.uris)
-    const postiveQuery = [
-      cts.jsonPropertyValueQuery('action', 'merge'),
-      urisQuery,
-      collectionQuery
-    ];
-    const otherMergesForURI = cts.search(
-      cts.andNotQuery(
-        cts.andQuery(postiveQuery),
-        uriRangeQuery
-      ),
-      ['unfiltered',cts.indexOrder(datahubCreatedOnRef, 'descending')]
-    ).toArray()
-        // get the actionNode for the URI
-        .map(
-          (result) => result.xpath(`/matchSummary/actionDetails/*[action = 'merge']`).toArray()
-            .filter((actionNode) => cts.contains(actionNode, urisQuery))[0]
-        )
-        // filter out false positives that don't have merge actions for the given URI
-        .filter((actionNode) => !!actionNode)
-        // convert node to JSON object
-        .map((actionNode) => actionNode.toObject());
-    if (otherMergesForURI.some((otherMerge) => otherMerge.uris.length > uriActionDetails.uris.length)) {
-      return Sequence.from([]);
+
+  let returnArray = [];
+
+  let thisMatchSummary = (content.value ? content.value : cts.doc(thisMatchSummaryURI));
+  if (!thisMatchSummary) {
+    return Sequence.from([]);
+  }
+  thisMatchSummary = thisMatchSummary.toObject();
+  let theseURIsToProcess = thisMatchSummary.matchSummary.URIsToProcess;
+
+  for (uriToProcess of theseURIsToProcess) {
+
+    // don't process this "URIsToProcess" URI unless we are processing the last matchSummary document
+    // (in URI order) that contains it
+    let lastSummaryWithURI = cts.uris(null, ["descending", "limit=1"],
+      cts.pathRangeQuery("/matchSummary/URIsToProcess", "=" , uriToProcess));
+    if (lastSummaryWithURI != thisMatchSummaryURI) {
+      continue;
     }
-    otherMergesForURI.forEach((actionDetails) => actionDetails.uris.forEach(
-      (uri) => {
-        if (!uriActionDetails.uris.includes(uri)) {
-          uriActionDetails.uris.push(uri);
-        }
+
+    let uriActionDetails = thisMatchSummary.matchSummary.actionDetails[uriToProcess];
+    // If the action is merge, ensure we create the merge with the most URIs
+    if (uriActionDetails && uriActionDetails.action === 'merge') {
```

Comment:
I kind of follow the logic, but I'm not clear on the goal - it sounds expected that there may be summaries with multiple possible merges for the same URI, and we want to favor the  match summary that recommends merging the most URIs with the current URI we're processing?